### DISCUSSION TITLE
Usecache decorator overwriting cache entry every time

### DISCUSSION THREAD

**Discussion by paulsmk** [TypedProject/ts-express-decorators#1897] [Q&A]
I am using the @UseCache decorator on a service function with mongoose cache, which is correctly writing the data to the cache, but on every call of the decorated function with the same params it is overwriting the cache (under the same key). Is this expected ? Should I be using cache.get to check for a cached entry before calling the decorated function ?

Thanks for the help.

**Comment by Romakita** [comment]
Can you tell me if you use the decorator over an endpoint method or on service method?

You are on which version?

**Reply by paulsmk** [reply]
It's a service method on version 6.105.4. This is the descriptor 
```typescript
@UseCache({ttl: 172800,type: ShippingRate})
private async calculateFedexShippingRate(store:Store, carrier:Carrier, rateParams:ShippingRateParams):Promise<ShippingRate|null> 
```

**Reply by Romakita** [reply]
Ok I fixed an issue about the useCache. Can you try to update all packages please :)

**Reply by paulsmk** [reply]
Thanks Romakita what version should I be on ?

**Reply by Romakita** [reply]
The latest is the better option :)

**Reply by Romakita** [reply]
The fix was released on [v6.114.6](https://github.com/tsedio/tsed/releases/tag/v6.114.6)

**Reply by paulsmk** [reply]
Still have the same effect I am afraid. Every call overwrites the cache. I am testing by manually editing the cache value in mongo, which I expect to be returned in the next call of the decorated function, but instead it is always overwritten. I am now on version 6.114.8

**Reply by paulsmk** [reply]
Just to confirm the decorated function should return the cached value automatically ? I dont need to manually retrieve from cache ?

**Reply by Romakita** [reply]
The expected behavior is to return the cached value but i introduced a strategy to refresh cache content in background when refreshthreshold option is given.

I’m just surprised because, I use this decorator and it works as expected (but I use redis). The code isn’t called  when the cache is filled.

can you open an issue please ;)

**Reply by Romakita** [reply]
Maybe the problem is here:
https://github.com/tsedio/tsed/blob/da9b1f42e6324679ae5d2e1488c1131dced2ba13/packages/platform/platform-cache/src/interceptors/PlatformCacheInterceptor.ts#L82

You can try to debug the code by editing the node module. If you can investigate it help me, because I won’t be available for the next 4 days.

see you

**Reply by paulsmk** [reply]
I'll take a look, thanks for your help

**Reply by paulsmk** [reply]
The problem is with cache-manager-mongoose, looks like it was noticed in December but the pull request remains unmerged https://github.com/disjunction/node-cache-manager-mongoose/pull/11#pullrequestreview-972478424

**Reply by Romakita** [reply]
Haaa good to know :D. 
So itsn’t necessary to open issue ;)

**Reply by paulsmk** [reply]
No, I'll try to get it merged!

**Reply by paulsmk** [reply]
@Romakita looks like that has now been merged and published as version 1.0.1 - would you be able to update tsed version of mongoose cache manager ?

**Reply by Romakita** [reply]
You just need to update your dependencies ;)
